modules/certificate/certificate.service.js

```diff
@@ -1,39 +1,46 @@
-const Certificate = require('./certificate.model');
+const { randomInt } = require('crypto');
+
 const RuleError = require('../../errors/rule.error');
+const Certificate = require('./certificate.model');
+const {
+  roles: { USER, ADMIN, SUPERADMIN },
+} = require('../../consts');
 
 const {
   STATUS_CODES: { NOT_FOUND, BAD_REQUEST },
 } = require('../../consts/status-codes');
 const {
   CERTIFICATE_NOT_FOUND,
-  CERTIFICATE_ALREADY_EXISTS,
+  CERTIFICATE_NOT_VALID,
+  CERTIFICATE_HAVE_OWNER,
+  CERTIFICATE_IS_ACTIVE,
+  CERTIFICATE_IS_USED,
 } = require('../../error-messages/certificate.messages');
 
 class CertificatesService {
   async getAllCertificates(skip, limit, user) {
-    let filter;
+    let filter = {};
 
-    if (user.role === 'admin' || user.role === 'superadmin') {
-      filter = {};
-    } else {
-      filter = { createdBy: user.id };
+    if (user.role === USER) {
+      filter = { ownedBy: user._id };
     }
+
     const items = await Certificate.find(filter)
       .sort({ startDate: 1 })
       .limit(limit)
       .skip(skip)
       .exec();
 
-    const count = Certificate.find(filter).countDocuments();
+    const count = items ? items.length : null;
```

Comment:
count can be null? why not just 0?

Reply:
Fixed it.  findOne returns null, find returns []